### ISSUE TITLE
Rule `no-unused-vars` doesn’t detect unused JSDoc `@import` types

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno_lint#1449]
### Lint Name

`no-unused-vars`

### Code Snippet

```js
// @ts-check

/** @import { ReactNode } from "react" */
```

### Expected Result

There should be a lint error that the type import `ReactNode` is unused.

### Actual Result

No lint error.

### Additional Info

In a `.ts` module, this:

```ts
import type { ReactNode } from "react";
```

Results in this lint error:

```
`ReactNode` is never used
If this is intentional, alias it with an underscore like `ReactNode as _ReactNode` deno-lint(no-unused-vars)
```

So it has already been established that `no-unused-vars` is supposed to enforce unused type imports.

Setting the TypeScript config `compilerOptions.noUnusedLocals` to `true` causes a TypeScript error about unused JSDoc `@import` types, but I would like to not have to set that and be able to rely on Deno lint.

### Version

deno 2.4.2 (stable, release, aarch64-apple-darwin)
v8 13.7.152.14-rusty
typescript 5.8.3
